tutorials/e2e-huggingface-summarization/src/run_summarization/run.py

```diff
@@ -0,0 +1,377 @@
+import logging
+import os
+import sys
+from datasets import load_dataset, load_metric, load_from_disk
+from transformers import (
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer,
+    DataCollatorForSeq2Seq,
+    Seq2SeqTrainingArguments,
+    Seq2SeqTrainer,
+    HfArgumentParser,
+    IntervalStrategy,
+)
+
+import torch
+import nltk
+from datasets import load_dataset
+from dataclasses import dataclass, field
+from typing import Optional
+import numpy as np
+import mlflow
+from pynvml import *
+import time
+from azureml.core import Run
+from azureml.core.model import Model
+
+
+@dataclass
+class DataArgs:
```

Comment:
could you add comments here explaining why you're using this format for arguments?
explain this is a HuggingFace recommended approach, and 1 line explaining why/how this work?

Reply:
comments added for dataclass usage.